optimum/habana/transformers/trainer.py

```diff
@@ -1790,3 +1806,22 @@ def _move_model_to_device(self, model, device):
         # Moving a model to HPU disconnects the tied weights, so we have to retie them.
         if self.args.use_habana and hasattr(model, "tie_weights"):
             model.tie_weights()
+
+    def _zero_model_grad(self, model):
+        if hasattr(model, "_zero_grad_kwargs"):
+            model.zero_grad(**model._zero_grad_kwargs)
+        else:
+            # Optimization based on setting gradients to None (instead of zeroing them out) may only be used when gradients are not recorded using HPU graphs.
+            # HPU graphs rely on fixed tensors - setting gradients to None will enforce their re-allocation during the backward pass each step.
+            set_to_none = (
+                self.args.local_rank == -1
+                and not self.args.use_hpu_graphs_for_training
+                and self.args.distribution_strategy != "fastddp"
```

Comment:
Is this condition needed since if `self.args.local_rank == -1` is True, there is no distributed run?

Reply:
You are right, I'm sorry for that. Changing to:
```
            set_to_none = (
                self.args.local_rank == -1 or self.args.distribution_strategy == "ddp"
            ) and not self.args.use_hpu_graphs_for_training
```